### PR TITLE
Disable vsync by default for all backends

### PR DESCRIPTION
**Pull request by gyagp**
To get FPS not capped by 60, we need to disable vsync. Use
--enable-vsync if you really need it.

### CODE REVIEW

File: src/aquarium-optimized/Aquarium.cpp [hunk 1]
```diff
@@ -349,15 +349,13 @@ bool Aquarium::init(int argc, char **argv) {
...
   }
 
-  if (result.count("turn-off-vsync")) {
-    if (!availableToggleBitset.test(
-            static_cast<size_t>(TOGGLE::TURNOFFVSYNC))) {
-      std::cerr << "Turn off vsync isn't supported for the backend."
-                << std::endl;
+  if (result.count("enable-vsync")) {
+    if (!availableToggleBitset.test(static_cast<size_t>(TOGGLE::ENABLEVSYNC))) {
+      std::cerr << "Enable vsync isn't supported for the backend." << std::endl;
```
Comment (hujiajie) [inline src/aquarium-optimized/Aquarium.cpp:354]: This is a little odd. Presumably vsync is more widely available than the other. IIRC, that's the status in early days of Dawn. If you insist on changing the default, what about:

* Keep the original toggle name `TURNOFFVSYNC`
* Add the toggle to `availableToggleBitset` if the backend supports turning off vsync, and set it in `toggleBitset` by default
* Reset/clear the toggle when `--enable-vsync` is given

Comment (gyagp) [reply src/aquarium-optimized/Aquarium.cpp:354]: I think for native benchmarks, it's more general to run with vsync off. For example, 3DMark is a famous native benchmark suite, and it suggests to set vsync off during tests. Actually "3DMark now warns you when vsync, FreeSync, or G-SYNC is enabled. For accurate results, you should disable these features in your video driver settings before benchmarking. " ( https://s3.amazonaws.com/download-aws.futuremark.com/3dmark-technical-guide.pdf, page 196).

Comment (hujiajie) [reply src/aquarium-optimized/Aquarium.cpp:354]: What if vsync is on in driver settings? Will 3DMark respect that by default but give a warning?

Comment (gyagp) [reply src/aquarium-optimized/Aquarium.cpp:354]: I think so, and then you will get the warning. But it's suggested to disable it before tests.

Comment (hujiajie) [reply src/aquarium-optimized/Aquarium.cpp:354]: Thanks. I still have some concerns on the naming of the toggle:

* If `ENABLE_FOO` can be optionally set in `availableToggleBitset`, it seems intuitive to say the opposite of feature `FOO` is supported by all backends, but it may be not the case for vsync.
* On the opposite, if the toggle is called `DISABLE_FOO`, then it sounds like feature `FOO` is enabled by default, which contradicts with the proposed patch here.

Comment (hujiajie) [reply src/aquarium-optimized/Aquarium.cpp:354]: Let me rephrase the proposals:

**A) Use the toggle name `ENABLEVSYNC` and disable vsync by default**

Here's how this can be misleading: if `ENABLEFOOBAR` is not set in `availableToggleBitset` → feature `FOOBAR` is not configurable in this backend → feature `FOOBAR` is not available/supported in this backend → the backend is expected to behave in a way similar to feature `FOOBAR` is disabled → substituting `FOOBAR` with vsync, the backend works as if vsync is disabled. However, when a backend doesn't set the `ENABLEVSYNC` bit in `availableToggleBitset`, it actually means the backend can only run in vsync mode and vsync cannot be disabled, which was the case in early days of Dawn.

**B) Use the toggle name `DISABLEVSYNC` and disable vsync by default**

Here's how this can be misleading: if `DISABLEFOOBAR` is not set in `toggleBitset` → it sounds like feature `FOOBAR` is enabled by default, and one needs to explicitly set this bit to disable the feature → substituting `FOOBAR` with vsync, vsync is enabled by default in backends. However, the proposal is to disable vsync by default!

**C) Use the toggle name `DISABLESYNC` and enable vsync by default**

This is the strategy currently used in master, and it doesn't have the ambiguity described in A) and B). However, benchmarking is a little inconvenient because one needs to pass an additional commandline flag to disable vsync.

**D) Leave it as TBD**

If #189 is implemented, we don't need this bikeshedding at all. Though I don't think it can happen anytime soon.

@gyagp @shaoboyan @Jiawei-Shao and anyone watching on this, your feedback is highly appreciated!
